app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+                    return convertView;
+                }
+            };
+
+            builder.setAdapter(adapter, (dialog, which) -> {
+                String userSelect = SYNC_PROVIDER_CHOICE_GPODDER_NET;
+                switch (which) {
+                    case 0:
+                        break;
+                    case 1:
+                        userSelect = SyncService.SYNC_PROVIDER_CHOICE_NEXTCLOUD;
+                        break;
+                    default:
+                        break;
+                }
+                setSelectedSyncProvider(userSelect);
+                if (userSelect.equals(SYNC_PROVIDER_CHOICE_GPODDER_NET)) {
+                    new GpodderAuthenticationFragment()
+                            .show(getChildFragmentManager(), GpodderAuthenticationFragment.TAG);
+                    updateScreen();
+                    return;
+                }
+                openNextcloudAccountChooser();
+                updateScreen();
+            });
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+            return true;
+        });
+        findPreference(PREF_GPODNET_SETLOGIN_INFORMATION)
+                .setOnPreferenceClickListener(preference -> {
+                    AuthenticationDialog dialog = new AuthenticationDialog(activity,
+                            R.string.pref_gpodnet_setlogin_information_title, false, GpodnetPreferences.getUsername(),
+                            null) {
+
+                        @Override
+                        protected void onConfirmed(String username, String password) {
+                            GpodnetPreferences.setPassword(password);
+                            SyncService.setIsProviderConnected(getContext(), true);
+                        }
+                    };
+                    dialog.show();
+                    return true;
+                });
+        findPreference(PREFERENCE_SYNC).setOnPreferenceClickListener(preference -> {
+            SyncService.syncImmediately(getActivity().getApplicationContext());
+            return true;
+        });
+        findPreference(PREFERENCE_FORCE_FULL_SYNC).setOnPreferenceClickListener(preference -> {
+            SyncService.fullSync(getContext());
+            return true;
+        });
+        findPreference(PREFERENCE_LOGOUT).setOnPreferenceClickListener(preference -> {
+            GpodnetPreferences.logout();
+            Snackbar.make(getView(), R.string.pref_gpodnet_logout_toast, Snackbar.LENGTH_LONG).show();
+            SyncService.setIsProviderConnected(getContext(), false);
+            setSelectedSyncProvider(SYNC_PROVIDER_UNSET);
+            updateScreen();
+            return true;
+        });
+    }
+
+    private void setSelectedSyncProvider(String userSelect) {
+        SharedPreferences preferences = getActivity()
+                .getSharedPreferences(SyncService.SHARED_PREFERENCES_SYNCHRONIZATION, Activity.MODE_PRIVATE);
```

Comment:
I think it's a bit strange that you write to the preferences object of the `SyncService` class from outside of the class. How about renaming `GpodnetPreferences` to simply `SyncPreferences` and storing the provider choice (and whatever else is needed, like user name) there?

Reply:
I wont touch GpodnetPreferences since it only deals with Gpodder.net sync provider and is totally independent from anything else.
Every synchronization provider should have its shared preference read/write logic isolated from the others.

I moved reading/writing to synchronization service preferences to the _SyncService_ class